lib/view/list_widget.dart

```diff
@@ -25,8 +25,8 @@ class _ListWidgetState extends State<ListWidget> {
   @override
   void initState() {
     super.initState();
-    listModel.reload();
     itemsToBeDisplayed = listModel.itemsView();
+    listModel.updateTimersForScheduledItems(timerCallback: _onItemEdited);
```

Comment:
I think you added this so that the list widget can be kept in sync with the list model when a timer completes?

A much better way to do this would be to create some kind of `Stream` inside the list model, that the list model adds events to when an item timer triggers. Then listen to the stream here and update the UI when an item timer triggers.

Reply:
> I think you added this so that the list widget can be kept in sync with the list model when a timer completes?

Yeah.

>A much better way to do this would be to create some kind of Stream inside the list model, that the list model adds events to when an item timer triggers. Then listen to the stream here and update the UI when an item timer triggers.

Oh, why didn't I think of that! 

Thank you for patiently teaching me.

Reply:
> ...that the list model adds events to when an item timer triggers

Thinking this through, two different ways to do this that spring to mind:
1. We could change the timer callback to both call `update` and trigger the event. So instead of:
```
timerCallback: update
```
  we would have something like:
```
timerCallback: (item) async {
  await update(item);
  addTimerTriggeredEventToStream();
}
```
2. We could trigger an event in the `update` function (possibly also in the `add` and `remove` functions). Something like:
```
Future<void> update(Item item) async {
  if (items.contains(item)) {
    // ...
    addItemUpdatedEventToStream();
  } else {
    // ...
  }
}
```

I am leaning towards the second option. Your thoughts?

Reply:
I think strategy 2 is better, with also `add` and `remove` triggering appropriate items in the stream.

Reply:
Basically all three methods `add`, `update`, `remove` should:
1. Do the operation
2. Add an event to the stream to notify the UI of the change